chore: add support for Release Candidates

NiccoloFei opened the pull request:
This PR enhances the release process to add support for release candidate (RC) versions.
Initially, only the YAML manifest installation method is going to be supported.

Closes #4778

NiccoloFei committed:
chore: add support for release candidates

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
ci: set the latest tag only on the latest stable image

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
docs: initial draft

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
docs: minor fix

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

gbartolini committed:
docs: explain RC

Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>

gbartolini committed:
fix: cosmetic changes

Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>

mnencia commented on .github/workflows/release-tag.yml:
```diff
@@ -21,6 +21,6 @@ jobs:
         with:
           token: ${{ secrets.REPO_GHA_PAT }}
-          version_regex: '^Version tag to ([0-9]+\.[0-9]+\.[0-9]+)'
+          version_regex: '^Version tag to ([0-9]+\.[0-9]+\.[0-9\-a-z-A-Z]+)'
```
```suggestion
          version_regex: '^Version tag to ([0-9]+\.[0-9]+\.[0-9]+(?:-[a-z][0-9a-z]*)?)'
```

NiccoloFei committed:
chore: allow releasing RCs from main branch

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
fix: adjust the version tag regexp

Co-authored-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
docs: minor changes

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

fcanovai committed:
ci: fix regex in conditional

The regex wasn't considering an initial v, and would have considered
every release a candidate one.

Signed-off-by: Francesco Canovai <francesco.canovai@enterprisedb.com>

sxd commented:
/ok-to-merg updating workflow

sxd commented:
/ok-to-merge updating workflow
